packages/office-ui-fabric-react/src/components/TextField/TextField.types.ts

```diff
@@ -159,7 +159,12 @@ export interface ITextFieldProps extends React.AllHTMLAttributes<HTMLInputElemen
   errorMessage?: string;
 
   /**
-   * Callback for the onChanged event.
+   * Callback for when the input value changes.
+   */
+  onChange?: (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => void;
```

Comment:
Is this the right change? 

The issue mentions `preventDefault` for example, that won't do anything here, all the action did already happen. 

`onChanged` was clear to me, if was fired after the change, here, just from looking at the props, what is the relationship between `onBeforeChange` and `onChange`? What comes first? Why does one give me the original event, and one doesn't? 

Does it make sense to expose the underlying html control or is that an implementation detail and we should change the handler to pass an instance of the `TextField` + the value?

Reply:
In general, sometimes the value is optional and a string, sometimes it's `any`, sometimes it's explicit `string | undefined`, sometimes (validation) it's just a string. What should be the right way to type the value going forward?

Reply:
We wanted to change the name to `onChange` and expose the event since that is the expected behavior of an HTML change handler function. `onBeforeChange` might need a similar change to expose the event in addition to the value - @dzearing is this a change we want to make?

Reply:
The value parameter has type `string` in the TextField `_onInputChange` function.